src/main/webapp/script.js

```diff
@@ -40,6 +41,13 @@ function displayError() {
 function getZipCode() {
   let zip = document.getElementById('zipCode').value;
   fetchList('/data?zipCode=' + zip);
+  initiateLoader();
+}
+
+function initiateLoader() {
+  document.getElementById('loader').style.display = 'block';
+  document.getElementById('loader-container').style.display = 'block';
```

Comment:
This might be unnecessary, but you could create global variables for loader and loader-container, and just call the .style.display on those. Might make the individual functions a bit clearer and avoid having to use the getElementById function each time.

Reply:
Done